Sources/Request/Request/RequestBuilder.swift

```diff
@@ -10,35 +10,26 @@ import Foundation
 @_functionBuilder
 public struct RequestBuilder {
     public static func buildBlock(_ params: RequestParam...) -> RequestParam {
-
-        let resultParams = params.filter { $0.children == nil } + params.compactMap { $0.children }.joined()
-        // Multiple Urls
-        if resultParams.filter({ $0.type == .url }).count > 1 {
-            fatalError("You cannot specify more than 1 `Url`")
-        }
-        // Missing Url
-        if resultParams.filter({ $0.type == .url }).count < 1 {
-            fatalError("You must have a `Url`")
-        }
-        return CombinedParams(children: resultParams)
+        CombinedParams(children: params)
     }
```

Comment:
I don't think we need `CombinedParams` or `unzip` if we just return an array of params in these builder functions:
```suggestion
    public static func buildBlock(_ params: RequestParam...) -> [RequestParam] {
        params
    }
```

Reply:
The functionBuilder `buildBlock(_:)` needs that the input parameter type should be equal to the return type, so I guess that this solution will not work. But, I would like to suggest that we remove the unzip getter and implement the `buildParam(_:)` on `CombinedParams`.

Reply:
Making the return type an array of `RequestParam`s actually makes the input and output match, as the input is an array of params. This is a valid function builder:
```swift
@_functionBuilder
struct ArrayBuilder<T> {
    static func buildBlock(_ items: T...) -> [T] {
        items
    }
}
```
I think implementing our function builder like this will simplify it quite a bit. You may need to return a single-item array of `RequestParam`s from the other build functions too.

Reply:
Still, it may work for simple function builders. In case that we have a `buildBlock(_:)` with an `ifBlock(_:)`, this will mess up the types used by the function builder, since an Array of RequestParam doesn't conform to RequestParam.

```
Request {
    Url("https://example.com")
    Query(["message": "Hello World"]
    // Until now RequestBuilder will only execute buildBlock(_:) resulting in [RequestParam] type

    if true {
        Body(["key": "value"])
        // will execute buildBlock(_:) resulting [RequestParam]
    }
    // ifBlocks needs a RequestParam? type not a [RequestParam]?
    // Now, merging the ifBlocks with buildBuild(_:) will be very tricky. 
    // Because now it has to execute a function that will match this buildBock(_: [RequestParam]...)
    // Or this buildBlock(_: [RequestParam], _: RequestParam)
}
```

So, I would like to suggest to keep the CombinedParams and use the unzip getter for a while. I will search more about it and feel free to update this code after merging it.

Reply:
Thanks for clarifying. I agree about making `RequestParam` have a `buildParam`